test/test_index_string.cpp

```diff
@@ -962,56 +957,73 @@ namespace {
 StringData create_string_with_nuls(const size_t bits, const size_t length, char* tmp, Random& random)
 {
     for (size_t i = 0; i < length; ++i) {
-        tmp[i] = (bits & (1 << i)) == 0 ? '\0' : static_cast<char>(random.draw_int<int>(CHAR_MIN, CHAR_MAX));
+        bool insert_nul_at_pos = (bits & (1 << i)) != 0;
+        if (insert_nul_at_pos) {
+            tmp[i] = '\0';
+        } else {
+            // Avoid stray \0 chars, since we are already testing all combinations.
+            // All casts are necessary to preserve the bitpattern.
+            tmp[i] = static_cast<char>(static_cast<unsigned char>(random.draw_int<unsigned int>(1, UCHAR_MAX)));
+        }
     }
     return StringData(tmp, length);
 }
 
 } // anonymous namespace
 
 
-// Test for generated strings of length 1..16 with all combinations of embedded NUL bytes
+// Test for generated strings of length 1..8/16 with all combinations of embedded NUL bytes
 TEST_TYPES(StringIndex_EmbeddedZeroesCombinations, non_nullable, nullable)
 {
     constexpr bool nullable = TEST_TYPE::value;
 
-    // String index
-    ref_type ref = StringColumn::create(Allocator::get_default());
-    StringColumn col(Allocator::get_default(), ref, nullable);
-    const StringIndex& ndx = *col.create_search_index();
-
-    const size_t MAX_LENGTH = 16; // Test medium
-    char tmp[MAX_LENGTH];         // this is a bit of a hack, that relies on the string being copied in column.add()
-
-    for (size_t length = 1; length <= MAX_LENGTH; ++length) {
+#if TEST_DURATION == 0
+    for (int seed = 0; seed < 5; seed++) {
+#else 
+    for (int seed = 0; seed < 100; seed++) {
+#endif
+        // String index
+        ref_type ref = StringColumn::create(Allocator::get_default());
+        StringColumn col(Allocator::get_default(), ref, nullable);
+        const StringIndex& ndx = *col.create_search_index();
 
-        {
-            Random random(42);
-            const size_t combinations = 1 << length;
-            for (size_t i = 0; i < combinations; ++i) {
-                StringData str = create_string_with_nuls(i, length, tmp, random);
-                col.add(str);
+#if TEST_DURATION == 0
+        const size_t MAX_LENGTH = 8;
```

Comment:
Can you move this variable declaration up so that it merges with the above `ifdef` block? It would make it a bit easier to read.